handlers/main.yml

```diff
@@ -1,7 +1,7 @@
 ---
 
-- name: Restart zookeeper
+- name: restart zookeeper
   systemd:
     name: zookeeper
     state: restarted
-  when: zookeeper_service_state != 'stopped'
+  when: zookeeper_service_state != 'stopped' and not zookeeper_service_output.changed
```

Comment:
what happens when the task that registers `zookeeper_service_output` has not been executed? Could this check be performed in the task?